### DISCUSSION TITLE
Does this app consume data of the internet provider?

### DISCUSSION THREAD

**Discussion by quazar-omega** [localsend/localsend#552] [Discussion (English, International)] [answered]
I.e. does the data go through a central server outside the local network to function, even after the connection has been established between the two devices? Or does it work like Magic Wormhole where that is done as a fallback?

**Comment by Tienisto** [comment, marked as answer]
No. Not a single byte is sent outside of the local network.

**Reply by quazar-omega** [reply]
Ok, thank you very much for clarifying!

**Reply by bagajohny** [reply]
@Tienisto 
I want some clarification on this. Suppose I connect my Windows laptop to the Personal Hotspot of my iPhone and transfer the files, will it still not use the mobile data from my data pack? And will it be the case if I send files from either of the device?
I am asking since iPhone does not all creating Personal Hotspot **unless you turn on mobile data connection**. Android allows this so this is not a concern on Android.

**Reply by quazar-omega** [reply]
Just my guess because I don't have an iPhone, but it most likely won't.  
The mobile data requirement is a mechanism built into iOS, but then, since the application doesn't route anything outside the local network, the data will just be sent directly and won't pass through any relay.  
You could test the data consumption by sending some sizable file and check if the quota has decreased significantly in a number close to the file size

**Reply by bagajohny** [reply]
Yeah I will eventually try this. I just wanted to know from the dev himself. Although I do have WiFi at home so it is not an immediate concern, in future I might face a situation where there is not wifi availability. This is the reason I wanted to know about it.

**Reply by Tienisto** [reply]
Unfortunately, I don't own an iPhone so I cannot test it myself.

LocalSend does send and receive data within the local network. It scans the local network to discover other devices. The scanning might be counted as traffic by your service provider.

Even if it counts, it should be negligible because the actual file transfer is peer-to-peer and only sent within the hotspot.
